src/core/atomic.d

```diff
@@ -1143,6 +1143,15 @@ version (CoreUnittest)
         }
     }
 
+    @betterC pure nothrow @nogc unittest{
+        shared byte* sptr = cast(byte*) 8;
+        byte* ptr = cast(shared byte*) 9;
+        assert(atomicFetchAdd(sptr, 1) == cast(shared byte*) 8);
+        assert(atomicFetchAdd(ptr, 1) == cast(byte*) 9);
+        assert(sptr == cast(shared byte*)9);
+        assert(ptr == cast(byte*)10);
```

Comment:
Please also add a test for pointers to non single byte objects. You can do it without using casts like this:

```D
shared(ulong)[] array = [2, 4, 6, 8, 10, 12, 14, 16, 19, 20];
{
    shared ulong* ptr = &array[0];
    shared(ulong)* prevPtr = atomicFetchAdd(ptr, 3); 
    assert(prevPtr == &array[0]);
    assert(*prevPtr == 2);
    assert(*ptr == 8);
}
{
    shared ulong* ptr = &array[5];
    shared(ulong)* prevPtr = atomicFetchSub(ptr, 4); 
    assert(prevPtr == &array[5]);
    assert(*prevPtr == 12);
    assert(*ptr == 4); // https://issues.dlang.org/show_bug.cgi?id=21578
}
```